syncmaster/worker/handlers/file/sftp.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import tempfile
+from typing import TYPE_CHECKING
+
+from onetl.connection import SFTP, SparkLocalFS
+from onetl.file import FileDFReader, FileDFWriter, FileDownloader, FileUploader
+
+from syncmaster.dto.connections import SFTPConnectionDTO
+from syncmaster.worker.handlers.file.base import FileHandler
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame, SparkSession
+
+
+class SFTPHandler(FileHandler):
+    connection_dto: SFTPConnectionDTO
+
+    def connect(self, spark: SparkSession) -> None:
+        self.connection = SFTP(
+            host=self.connection_dto.host,
+            port=self.connection_dto.port,
+            user=self.connection_dto.user,
+            password=self.connection_dto.password,
+            compress=False,  # to avoid errors from combining file and SCP-level compression
+        ).check()
+        self.local_connection = SparkLocalFS(
+            spark=spark,
+        ).check()
+
+    def read(self) -> DataFrame:
+        from pyspark.sql.types import StructType
+
+        with tempfile.TemporaryDirectory(prefix="syncmaster_sftp_") as temp_dir:
+            downloader = FileDownloader(
+                connection=self.connection,
+                source_path=self.transfer_dto.directory_path,
+                local_path=temp_dir,
+            )
+            downloader.run()
+
+            reader = FileDFReader(
+                connection=self.local_connection,
+                format=self.transfer_dto.file_format,
+                source_path=temp_dir,
+                df_schema=StructType.fromJson(self.transfer_dto.df_schema) if self.transfer_dto.df_schema else None,
+                options=self.transfer_dto.options,
+            )
+            df = reader.run()
+            df.cache().count()
```

Comment:
Hm, I see, dataframe is created over temporal directory, so downloaded files are dropped immediately after df is created.

Instead, we can create temp directory outside the handler class, and then pass it as constructor argument.